akka-remote/src/test/scala/akka/remote/artery/tcp/TlsTcpSpec.scala

```diff
@@ -171,3 +174,40 @@ class TlsTcpWithHostnameVerificationSpec extends ArteryMultiNodeSpec(
     }
   }
 }
+
+class TlsTcpWithActorSystemSetupSpec
+  extends ArteryMultiNodeSpec(TlsTcpSpec.config) with ImplicitSender {
+
+  val sslProviderServerProbe = TestProbe()
+  val sslProviderClientProbe = TestProbe()
+  val sslProviderSetup = SSLEngineProviderSetup(sys ⇒ new ConfigSSLEngineProvider(sys) {
+    override def createServerSSLEngine(hostname: String, port: Int): SSLEngine = {
+      sslProviderServerProbe.ref ! "createServerSSLEngine"
+      super.createServerSSLEngine(hostname, port)
+    }
+
+    override def createClientSSLEngine(hostname: String, port: Int): SSLEngine = {
+      sslProviderClientProbe.ref ! "createClientSSLEngine"
+      super.createClientSSLEngine(hostname, port)
+    }
+
+  })
+
+  val systemB = newRemoteSystem(name = Some("systemB"), setup = Some(ActorSystemSetup(sslProviderSetup)))
+  val addressB = address(systemB)
+  val rootB = RootActorPath(addressB)
+
+  "Artery with TLS/TCP with SSLEngineProvider defined via Setup" must {
+    "use the right SSLEngineProvider" in {
+      systemB.actorOf(TestActors.echoActorProps, "echo")
+      val path = rootB / "user" / "echo"
+      system.actorSelection(path) ! Identify(path.name)
+      val echoRef = expectMsgType[ActorIdentity].ref.get
+      echoRef ! "ping-1"
+      expectMsg("ping-1")
+
+      sslProviderServerProbe.expectMsg("createServerSSLEngine")
```

Comment:
Do we have a similar test for Netty?

Reply:
The Setup is only supported by Artery. Didn't want to add to much to classic.